staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -181,6 +181,20 @@ type VolumeSource struct {
 	//
 	// +optional
 	Ephemeral *EphemeralVolumeSource `json:"ephemeral,omitempty" protobuf:"bytes,29,opt,name=ephemeral"`
+	// image represents a OCI object (image or artifact) pulled and mounted on the kubelet's host machine.
+	// The volume is resolved at pod startup depending on which PullPolicy value is provided:
+	//
+	// - Always: the kubelet always attempts to pull the reference. Container creation will fail If the pull fails.
+	// - Never: the kubelet never pulls the reference and only uses a local image or artifact. Container creation will fail if the reference isn't present.
+	// - IfNotPresent: the kubelet pulls if the reference isn't already present on disk. Container creation will fail if the reference isn't present and the pull fails.
+	//
+	// The volume gets re-resolved if the pod gets recreated, which means that new remote content will become available on pod recreation.
```

Comment:
Trying to think about whether to add any additional description, but this is pretty succinct.  

We don't really define the retry behavior of volume start well, but given that this one is likely to add both latency and more failure causes, we should probably add some content along those lines as an additional sentence:

> A failure to resolve or pull the image during pod startup will block containers from starting and may add significant latency.  Failures will be retried using normal volume backoff and will be reported on the pod reason and message. [need to verify that is true for this change]

It would be nice (depending on how complex to implement) if failures show up in both the PodInitialized condition.  It's probably required for those errors to show up in pod.status.reason / message (I think they do today for pull failures but would like to double check during implementation).

Reply:
Yes, we're pulling in the same manner during pod sync and will report the same errors. I updated the docs to reflect that.